streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -440,6 +440,18 @@ public void commit() {
         commit(true);
     }
 
+    private long getNextPartitionTime(final TopicPartition partition) {
+        final OffsetAndMetadata oldCommittedMetadata = consumer.committed(partition);
+        final long metadataTimestamp;
+        if (oldCommittedMetadata != null && oldCommittedMetadata.metadata().length() != 0) {
+            metadataTimestamp = Long.parseLong(oldCommittedMetadata.metadata());
+        } else {
+            metadataTimestamp = RecordQueue.UNKNOWN;
+        }
+        final long localPartitionTime = partitionTime(partition);
+        return localPartitionTime < metadataTimestamp ? metadataTimestamp : localPartitionTime;
```

Comment:
would be favorable to order comparison result according to first citizen. Like
`metadataTimestamp >= localPartitionTime ? metadataTimestamp : localPartitionTime;`

Reply:
Well, the order in general doesn't seem to matter that much. But could change it.